internal/prune/graphList.go

```diff
@@ -3,44 +3,59 @@ package prune
 import (
 	"bytes"
 	"fmt"
+	"github.com/gleanerio/nabu/internal/graph"
 	"github.com/gleanerio/nabu/pkg/config"
 	"github.com/minio/minio-go/v7"
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 	"github.com/tidwall/gjson"
-	"io/ioutil"
+	"io"
 	"net/http"
 	"net/url"
 	"strings"
 )
 
 func graphList(v1 *viper.Viper, mc *minio.Client, prefix string) ([]string, error) {
+	log.Println("Getting list of named graphs")
+
 	ga := []string{}
 
-	spql, _ := config.GetSparqlConfig(v1)
-	bucketName, _ := config.GetBucketName(v1)
+	ep := v1.GetString("flags.endpoint")
+	spql, err := config.GetEndpoint(v1, ep, "sparql")
+	if err != nil {
+		log.Error(err)
+	}
+
+	bucketName, err := config.GetBucketName(v1)
+	if err != nil {
+		log.Println(err)
+		return ga, err
+	}
 
-	// Reference ADR 0001 for why we are building the regex here for the graphs like this.
-	s2c := strings.Replace(prefix, "summoned/", ":", -1)
-	gp := fmt.Sprintf("urn:%s%s:", bucketName, s2c)
-	fmt.Printf("Pattern: %s\n", gp)
+	gp, err := graph.MakeURNPrefix(v1, bucketName, prefix)
+	if err != nil {
+		log.Println(err)
+		return ga, err
+	}
 
 	d := fmt.Sprintf("SELECT DISTINCT ?g WHERE {GRAPH ?g {?s ?p ?o} FILTER regex(str(?g), \"^%s\")}", gp)
```

Comment:
thought: query for graphs, and filter in go

21 seconds, 343164 named graphs

```
SELECT DISTINCT ?g 
WHERE {
  GRAPH ?g { ?s ?p ?o }
}
```

120 seconds, 18 named graphs 
`SELECT DISTINCT ?g WHERE {GRAPH ?g {?s ?p ?o} FILTER regex(str(?g), "^urn:test:eco:geocodes_demo_datasets")}`

Reply:
I'll update the code for this...